src/dbnode/storage/index/block.go

```diff
@@ -965,6 +965,11 @@ func (b *block) addQueryResults(
 	results BaseResults,
 	batch []doc.Document,
 ) ([]doc.Document, int, error) {
+	// update recently queried docs to monitor memory.
+	if err := b.opts.QueryStats().Update(len(batch)); err != nil {
```

Comment:
May want to cache `b.queryStats = b.opts.QueryStats() on the block construction (i.e. NewBlock fn) so this function call is not required each time we add a batch of index document results to the results set. (then it's just `b.queryStats.Update(len(batch))` here).